README.md

```diff
@@ -1,240 +1,268 @@
+   ```
+2. Run a container using the tagged image.
+   ```
+   docker run kartograafr
+   ```
 
-The file system in the instance is volatile.  The application can read
-and write to the local file system but a restart will make those
-changes vanish.  Docker *can* easily be configured to map external
-persistant disk to internal file system directories but that must be
-configured explicitly when invoking Docker.
+Note: Currently, there is no way to test sending emails using Docker on a local machine.
 
 ### OpenShift
 
-Applications that run under Docker are usually easy to run on
-OpenShift.  There will be configuration changes.  OpenShift
-configuration is beyond the scope of this readme.  The *runDocker.sh*
-script is a model for what OpenShift needs to be configured to supply.
-
-------
-
-###  Secure Information
-
-Any secure or authentication information needs to be
-kept out of source control and be strictly separated from normal
-properties.  That information is expected to be found in the
-*secrets.py* module which will be available in a separate secrets directory.
-A template for a secrets file is in the the configuration directory.
-That template shouldn't be modified but should be copied and that copy
-should be read from the separate secure directory.
-
-When running locally the *startup.sh* script assumes that the
-secrets.py file is available in the subdirectory *./OPT/SECRETS*.
-That reference can be a symbolic link to the actual location.  On
-OpenShift the location will be set in the deployment
-configuration. See the OpenShift kartograafr dev project to see how it
-is configured.
-
-The template file illustrates how to easily implement an environmental variable
-override of the property value.  The code given is very likely not the
-best way to do this and is likely to be improved soon.
+Deploying the application as a job using OpenShift and Jenkins involves several steps, which are beyond the scope of
+this README. However, it seems appropriate to explain some configuration steps assumed by the `Dockerfile` and unique 
+to this application.
+
+* The `env.json` file described in the **Application** section under Configuration needs to be made available to 
+  running kartograafr containers via an OpenShift ConfigMap, a type of Resource. A volume containing the ConfigMap 
+  should be mapped to the `configuration/secrets` subdirectory. These details will be specified in a configuration file
+   (.yaml) defining the pod.
+
+* The kartograafr application writes log files as part of its normal operations. Because of permission restrictions 
+  within OpenShift pods, users may need to specify a path for `Logging_Directory` in `env.json` that begins with
+  `tmp`. The U-M configuration uses `tmp/log/kartograafr`. Note that these log files will disappear with the running
+  container.
+
+* By default, the application will run without sending email and with the assumption that the JSON file will be named 
```

Comment:
Is there a sequence for checking the "send email" setting? e.g. env.json ->env.yaml -> other file? And maybe the last wins?

Reply:
Right now, the only way to indicate that you want to send email in OpenShift is by specifying an environment variable. The `start.sh` script indicated as the starting point in the `Dockerfile` checks whether the `SEND_EMAIL` value is specified as `"True"`. If it is assigned that string, it runs with email; if it isn't specified at all or specified as any other string, it will run without email. The Shell script just runs the `main.py` file with or without the `--email` flag. Adding a `Send_Email` variable in the `env.json` would make sense, but that would require code changes and having multiple ways of setting this value might be confusing (command-line arguments and in the config). Does that answer your question?